### PR TITLE
[next] temp parent refactor

### PR DESCRIPTION
**Pull request by ivanpopelyshev**
Another version of #4596

I agree with @englercj that we should not add this branching in critical place. Although transform itself has two branches inside, I just dont like that temporary parent hack in a critical place.

I can rename `safeUpdateTransform` if anyone has ideas about new name.

### CODE REVIEW

File: packages/display/src/DisplayObject.js [hunk 1]
```diff
@@ -146,17 +145,38 @@ export default class DisplayObject extends EventEmitter
...
+        this.transform.updateTransform(parent.transform);
+        // color/alpha transform
+        this.worldAlpha = this.alpha * parent.worldAlpha;
+    }
+
+    /**
+     * Updates the object transform (coordinates and alpha) for rendering
+     * even if parent is null
+     */
+    safeUpdateTransform()
```
Comment (bigtimebuddy) [inline packages/display/src/DisplayObject.js]: Couldn't this all be simplified to and remove the if/else:
```js
const parent = this.parent;
this.parent = parent || this._tempDisplayObjectParent;
this.updateTransform();
this.parent = parent;
```

File: packages/display/src/DisplayObject.js [hunk 2]
```diff
@@ -266,20 +277,6 @@ export default class DisplayObject extends EventEmitter
         if (!skipUpdate)
         {
             this._recursivePostUpdateTransform();
```
Comment (bigtimebuddy) [inline packages/display/src/DisplayObject.js]: Don't you need to call `this.safeUpdateTransform();` here?

File: packages/display/src/DisplayObject.js [hunk 3]
```diff
@@ -306,20 +303,6 @@ export default class DisplayObject extends EventEmitter
...
```
Comment (bigtimebuddy) [inline packages/display/src/DisplayObject.js]: What about here too?

File: packages/display/src/DisplayObject.js (hunk 2 above)
Comment (ivanpopelyshev) [reply packages/display/src/DisplayObject.js]: Nope, its not required for toLocal/toGlobal. there was single displayObjectUpdateTransform, without children, but its called from _recursivePostUpdateTransform anyway!

File: packages/display/src/DisplayObject.js (hunk 3 above)
Comment (ivanpopelyshev) [reply packages/display/src/DisplayObject.js]: ^^^ same.

File: packages/display/src/DisplayObject.js (hunk 2 above)
Comment (bigtimebuddy) [reply packages/display/src/DisplayObject.js]: Cool, just making sure

**Comment by bigtimebuddy** [comment]
I'm not the biggest fan of the name `safeUpdateTransform`.  I think we need to be more explicit than just "safe" about what this is doing. First question, does this need to be a public method?

File: packages/display/src/DisplayObject.js (hunk 1 above)
Comment (ivanpopelyshev) [reply packages/display/src/DisplayObject.js]: I think its making code less verbose.

**Comment by ivanpopelyshev** [comment]
Its one of those plugin-level methods. I agree that it has to be renamed.

File: packages/display/src/DisplayObject.js (hunk 1 above)
Comment (cursedcoder) [reply packages/display/src/DisplayObject.js]: @bigtimebuddy you can't modify `const` value :)

I would give +1 for `if-else` statement, it improves readability at least twice

File: packages/display/src/Container.js [hunk 4]
```diff
@@ -315,15 +315,17 @@ export default class Container extends DisplayObject
 
     /**
      * Updates the transform on all children of this container for rendering
+     * parent must exist
```
Comment (cursedcoder) [inline packages/display/src/Container.js]: missing dot here

File: packages/display/src/Container.js [hunk 5]
```diff
@@ -315,15 +315,17 @@ export default class Container extends DisplayObject
...
-
-        this.transform.updateTransform(this.parent.transform);
+        const parent = this.parent;
 
-        // TODO: check render flags, how to process stuff here
-        this.worldAlpha = this.alpha * this.parent.worldAlpha;
+        this._boundsID++;
+        // coords transform
+        this.transform.updateTransform(parent.transform);
+        // color/alpha transform
```
Comment (cursedcoder) [inline packages/display/src/Container.js]: doesn't 325 and 327 comments seem obvious?

**Comment by bigtimebuddy** [comment]
I'm going to close this because of inactivity, no consensus around naming the new API, and the lack of a clearly-defined performance to understand benefits. If someone can describe the performance benefit, will consider avoiding the extra branch.

**Pull request closed without merging**